libs/template/materialize.go

```diff
@@ -40,6 +40,13 @@ func Materialize(ctx context.Context, configFilePath, templateRoot, outputDir st
 		return err
 	}
 
+	if outputDir == "" {
```

Comment:
This is needed to always get the full absolute path

Reply:
This is not enough, since outputDir could be `.` or some other relative path. You could use `filepath.Abs` instead and directly fix this at the call site where we define a new `copyFile` or `inMemoryFile` to:
```
dstPath: &destinationPath{
root:    filepath.Abs(r.instanceRoot),
relPath: relPath,
},
```